nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py

```diff
@@ -588,6 +622,55 @@ def setup_test_data(self, cfg):
             )
             self._test_dl = self.build_pretraining_data_loader(self._test_ds, consumed_samples)
 
+    def configure_optimizers(self):
+        retval = super().configure_optimizers()
+
+        if self.with_distributed_adam:
```

Comment:
Is this monkey patch really necessary? Can't we ask apex code to support our use case ? This method seems incredibly fragile.

Reply:
I agree it's not a nice solution. I think there are three ways to support this feature:
- Monkey-patch this function
- Add Megatron-specific code into the Apex class
- Create a sub-class of the optimizer within NeMo and override this function

Now that I think about it, the third option sounds more reasonable.

Reply:
3rd option definitely seems like the way to go.